modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -774,14 +774,19 @@ class PrePostProcWrapper {
         if (explicit_in_model_layout) {
             input_info.model().set_layout(::ov::Layout(*explicit_in_model_layout));
         } else if (m_model->input(input_name).get_shape().size() == 4u) {
-            // NB: Back compatibility with IR's without any layout information.
-            // Note that default is only applicable for 4D inputs in order to
-            // support auto resize for image use cases.
-            GAPI_LOG_WARNING(NULL, "Failed to find layout for input layer \""
-                    << input_name << "\" - NCHW is set by default");
-            const std::string default_layout = "NCHW";
-            input_info.model().set_layout(::ov::Layout(default_layout));
-            m_input_model_layout.emplace(input_name, default_layout);
+            const auto& input_layout = ::ov::layout::get_layout(m_model->input(input_name));
+            if (!input_layout.empty()) {
+                input_info.model().set_layout(input_layout);
```

Comment:
Looking at it on the second day, is this a valid change at all? Why we're setting _whole_ model's layout based on _some_ input layout?

Reply:
It's not for the `whole` model, it's set to `input_info` which refers to particular input layer.
```
input_info = m_ppp.input(input_name);
...
input_info.model().set_layout(input_layout);
```